tonic/transforms.py

```diff
@@ -121,6 +121,95 @@ def __call__(self, events):
         return functional.drop_event_numpy(events, self.p, self.random_p)
 
 
+@dataclass(frozen=True)
+class DropEventByTime:
+    """Drops events in a certain time interval with a length proportional to a specified ratio of the original length.
+
+    Parameters:
+        duration_ratio (Union[float, Tuple[float]], optional): the length of the dropped time interval, expressed in a ratio of the original sequence duration.
+            - If a float, the value is used to calculate the interval length
+            - If a tuple of 2 floats, the ratio is randomly chosen in [min, max)
+            Defaults to 0.2.
+        starts_at_zero (bool, optional): flag that indicates whether the input sequence starts at time-step 0.
+            If False, takes the earliest timestamp as start timestamp. Defaults to True.
+
+    Example:
+        >>> transform = tonic.transforms.DropEventByTime(duration_ratio=(0.1, 0.8), starts_at_zero=True)
+    """
+
+    duration_ratio: Union[float, Tuple[float]] = 0.2
+    starts_at_zero: bool = True
+
+    def __call__(self, events):
+
+        return functional.drop_by_time_numpy(
+            events, self.duration_ratio, self.starts_at_zero
+        )
+
+
+@dataclass(frozen=True)
+class DropEventByArea:
+    """Drops events located in a randomly chosen box area. The size of the box area is defined by a specified ratio of the sensor size.
+
+    Args:
+        sensor_size (Tuple): size of the sensor that was used [W,H,P]
+        area_ratio (Union[float, Tuple[float]], optional): Ratio of the sensor resolution that determines the size of the box area where events are dropped.
+            - if a float, the value is used to calculate the size of the box area
+            - if a tuple of 2 floats, the ratio is randomly chosen in [min, max)
+            Defaults to 0.2.
+
+    Example:
+        >>> transform = tonic.transforms.DropEventByArea(sensor_size=(128,128,2), area_ratio=(0.1, 0.8))
+    """
+
+    sensor_size: Tuple[int, int, int]
+    area_ratio: Union[float, Tuple[float]] = 0.2
+
+    def __call__(self, events):
+
+        return functional.drop_by_area_numpy(events, self.sensor_size, self.area_ratio)
+
+
+@dataclass(frozen=True)
+class EventDrop:
+    """Applies EventDrop transformation from the paper "EventDrop: Data Augmentation for Event-based Learning".
+        Applies one of the 4 drops of event strategies between:
+            1. Identity (do nothing)
+            2. Drop events by time
+            3. Drop events by area
+            4. Drop events randomly
+
+        For each strategy, the ratio of dropped events are determined in the paper.
+
+    Args:
+        sensor_size (Tuple): size of the sensor that was used [W,H,P]
+        starts_at_zero (bool, optional): flag that indicates whether the input sequence starts at time-step 0.
+            If False, takes the earliest timestamp as start timestamp. Defaults to True.
```

Comment:
I understand that if your recording started at zero, but there are no events in the beginning, you still might want to apply the temporal drop in the beginning for consistency with other recordings. However I do not understand under what circumstances I would ever set this to False? If I know that all my recordings start at time step 20, I can just remove that offset right?